### ISSUE TITLE
Multiple rollbacks don't work starting with 0.2.4

### ISSUE THREAD

**Issue by workmanw** [lytics/ember-data.model-fragments#56]
See this JSBIN: http://emberjs.jsbin.com/yemimobire/5/edit

This is working with 0.2.3, but broken with 0.2.4 and 0.2.6 (didn't bother checking 0.2.5).

=====

**Edit: Sorry for the poor description.**

The first time you call rollback on a model, it works correctly (restoring the original value and setting isDirty back to false), the second time you call rollback on that same model it sets isDirty back to false, but does no restore the value.

**Comment by slindberg** [comment]
Thanks for testing this in your app and reporting, it looks like this is happening [with `hasManyFragment` properties as well](http://emberjs.jsbin.com/xedunuwaza/1/edit?html,js,output). Sorry for all the issues with this upgrade, the shift away from observing the `data` property had a lot of unseen consequences. I'll take a look at this as soon as I can.

/cc @igorT

**Comment by workmanw** [comment]
Hey no worries at all! Happy to. It just so happened we had an integration test that caught this one.

**Comment by slindberg** [comment]
@workmanw would you mind testing the fix in your app using the `tmp-dist` branch? you can use [this file](https://rawgit.com/lytics/ember-data.model-fragments/tmp-dist/dist/ember-data.model-fragments.js) for jsbins or put this in your `bower.json`:

```json
{
  "dependencies": {
    "ember-data.model-fragments": "lytics/ember-data.model-fragments#tmp-dist",
    ...
  }
}
```

**Comment by workmanw** [comment]
:smile: Confirmed fix!

Just to note, I first upgrade to 0.2.6 to confirm that our tests would still fail (they did). Then I upgraded to your #tmp-dist branch and our tests passed. Thanks!!!

**Comment by slindberg** [comment]
Fixed in 0.2.7

**Comment by workmanw** [comment]
Thanks again!
